### DISCUSSION TITLE
Using upsert

### DISCUSSION THREAD

**Discussion by viveleroi** [prisma/prisma#16882] [Q&A]
I was hoping to use `upsert` to avoid having to create or update myself but the `where` condition will always use a MongoDB ObjectID, which will be undefined if the record doesn't exist. Our front-end uses a single page for both add and edit functions. It's only an "edit" when there's an ID, otherwise it's an "add".

Problem is, prisma throws an error when id is undefined because obviously that becomes an invalid query. Is there any way for me use upsert this way, or should I just use create and update.

```js
prisma.server.upsert({
      where: {
        id: input.id
      },
      update: input,
      create: {
        ...input,
        owner: {
          connect: {
            id: ctx.session.user.id
          }
        }
      }
    })
```

**Comment by pantharshit00** [comment]
You can probably workaround it by passing a default value:
```ts
prisma.server.upsert({
      where: {
        id: input.id ?? ObjectId.createFromTime(0).toString()
      },
      update: input,
      create: {
        ...input,
        owner: {
          connect: {
            id: ctx.session.user.id
          }
        }
      }
    })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
